ladybug/legend.py

```diff
@@ -0,0 +1,877 @@
+                'colors': [col.to_json() for col in self.colors],
+                'continuous_colors': self.continuous_colors,
+                'continuous_legend': self.continuous_legend, 'title': title,
+                'ordinal_dictionary': self.ordinal_dictionary,
+                'number_decimal_places': self.number_decimal_places,
+                'include_larger_smaller': self.include_larger_smaller,
+                'vertical_or_horizontal': self.vertical_or_horizontal,
+                'base_plane': base_plane,
+                'segment_height': seg_h, 'segment_width': seg_w,
+                'text_height': txt_h, 'font': self.font,
+                'is_number_of_segments_default': self.is_number_of_segments_default,
+                'is_title_default': self.is_title_default,
+                'is_base_plane_default': self.is_base_plane_default,
+                'is_segment_height_default': self.is_segment_height_default,
+                'is_segment_width_default': self.is_segment_width_default,
+                'is_text_height_default': self.is_text_height_default}
+
+    def __copy__(self):
+        new_par = LegendParameters(
+            self.min, self.max, self.number_of_segments, self.colors,
+            self.continuous_colors, self.continuous_legend, self.title,
+            self.ordinal_dictionary, self.number_decimal_places,
+            self.include_larger_smaller, self.vertical_or_horizontal,
+            self.base_plane, self.segment_height, self.segment_width, self.text_height,
+            self.font)
+        new_par._is_number_of_segments_default = self._is_number_of_segments_default
+        new_par._is_title_default = self._is_title_default
+        new_par._is_base_plane_default = self._is_base_plane_default
+        new_par._is_segment_height_default = self._is_segment_height_default
+        new_par._is_segment_width_default = self._is_segment_width_default
+        new_par._is_text_height_default = self._is_text_height_default
+        return new_par
+
+    def ToString(self):
+        """Overwrite .NET ToString method."""
+        return self.__repr__()
+
+    def __repr__(self):
+        """Legend parameter representation."""
+        min = self.min if self.min is not None else '[default]'
+        max = self.max if self.max is not None else '[default]'
+        seg = '[default]' if self.is_number_of_segments_default \
+            else self.number_of_segments
+        title = '[default]' if self.is_title_default else self.title
+        base_pt = '[default]' if self.is_base_plane_default else self.base_plane.o
+        seg_h = '[default]' if self.is_segment_height_default else self.segment_height
+        seg_w = '[default]' if self.is_segment_width_default else self.segment_width
+        txt_h = '[default]' if self.is_text_height_default else self.text_height
+        return 'Legend Parameters\n minimum: {}\n maximum: {}\n segments: {}\n' \
```

Comment:
Might be too much info for a quick print looks like a debugging output. If user needs to access a certain property they can just access that.

Reply:
I originally had it much leaner but I found myself constantly wanting to check defaults that were being assigned for all the properties, especially from Grasshopper.  I guess we can just have the ToString() method output this text and make the `__repr__` lighter. Does that sound like a better alternative and, if so, should the `__repr__` contain any of the properties?

Reply:
I think your work can be considered as being in "debug mode" but if you think it is useful let's keep it. I think `__repr__` and `ToString` be identical as they are technically the same.

Reply:
Ok. Glad to hear it. We can always change it if we realize it is too unwieldy.